backend/src/ee/services/secret-snapshot/secret-snapshot-service.ts

```diff
@@ -240,22 +246,37 @@ export const secretSnapshotServiceFactory = ({
             },
             tx
           );
-          const snapshotSecrets = await snapshotSecretV2BridgeDAL.insertMany(
-            secretVersions.map(({ id }) => ({
-              secretVersionId: id,
-              envId: folder.environment.envId,
-              snapshotId: newSnapshot.id
-            })),
-            tx
-          );
-          const snapshotFolders = await snapshotFolderDAL.insertMany(
-            folderVersions.map(({ id }) => ({
-              folderVersionId: id,
-              envId: folder.environment.envId,
-              snapshotId: newSnapshot.id
-            })),
-            tx
-          );
+
+          const chunkedSnapshotSecrets = chunkArray(secretVersions, 2500);
```

Comment:
Ohh your chunking like this. 

You can do this instead. Its in the ORM already.

https://github.com/Infisical/infisical/blob/09d28156f8e1a60cde9f31bd7db89502319c00b7/backend/src/lib/knex/index.ts#L144